common/src/edgeai_camera.cpp

```diff
@@ -0,0 +1,144 @@
+/*
+ *  Copyright (C) 2023 Texas Instruments Incorporated - http://www.ti.com/
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *    Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *
+ *    Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the
+ *    distribution.
+ *
+ *    Neither the name of Texas Instruments Incorporated nor the names of
+ *    its contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ *  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ *  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ *  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ *  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ *  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/* Module headers */
+#include <common/include/edgeai_camera.h>
+#include <utils/include/ti_logger.h>
+
+namespace ti::edgeai::common
+{
+
+using namespace ti::utils;
+
+camera::camera()
+{
+    LOG_DEBUG("camera CONSTRUCTOR\n");
+}
+
+camera::~camera()
+{
+    LOG_DEBUG("multiScaler DESTRUCTOR\n");
+}
+
+int32_t camera::getConfig(InputInfo* camInputInfo, int32_t chMask)
+{
+    vx_status status = VX_SUCCESS;
+
+    string viss_dcc_path = "/opt/imaging/" + camInputInfo->m_sen_id + "/dcc_viss_wdr.bin";
+    string ldc_dcc_path = "/opt/imaging/" + camInputInfo->m_sen_id + "/dcc_ldc_wdr.bin";
+
+    tiovx_sensor_module_params_init(&sensorObj);
+    sensorObj.is_interactive = 0;
+    sensorObj.ch_mask = chMask;
+
+    tiovx_sensor_module_query(&sensorObj);
+
+    /* App works only for IMX390 2MP cameras */
+    if(camInputInfo->m_sen_id == "imx390")
```

Comment:
Should support other cameras that the multicam/singlecam application has.

Reply:
I have now tested only with IMX390, so just going with that in this release. As and when I get the opportunity to test with other sensors, I will include them here.